src/@context/Pool/index.tsx

```diff
@@ -97,137 +89,120 @@ function PoolProvider({ children }: { children: ReactNode }): ReactElement {
+      if (
+        !poolData ||
+        !poolInfo?.totalPoolTokens ||
+        poolData.shares[0]?.shares === '0'
+      )
+        return
+
+      // Pool share tokens.
+      const poolSharePercentage = new Decimal(poolData.shares[0]?.shares)
+        .dividedBy(poolInfo.totalPoolTokens)
+        .mul(100)
+        .toFixed(2)
+
+      const ownerLiquidity = calcSingleOutGivenPoolIn(
+        poolData.baseTokenLiquidity,
+        poolData.totalShares,
+        poolData?.shares[0]?.shares
+      )
+
+      const newPoolOwnerInfo = {
+        liquidity: ownerLiquidity,
+        poolShares: poolData.shares[0]?.shares,
+        poolSharePercentage
+      }
+      setPoolInfoOwner(newPoolOwnerInfo)
+      LoggerInstance.log(
+        '[pool] Created new pool creatorinfo:',
+        newPoolOwnerInfo
+      )
     }
-    setPoolInfoOwner(newPoolOwnerInfo)
-    LoggerInstance.log('[pool] Created new owner pool info:', newPoolOwnerInfo)
+
+    calculatePoolOwnerInfo()
   }, [
-    ownerPoolShares,
+    asset.chainId,
     poolData,
-    poolInfo.totalLiquidityInOcean,
-    poolInfo.totalPoolTokens
+    poolInfo?.baseTokenAddress,
+    poolInfo?.totalPoolTokens
   ])
 
   //
   // 3 User Pool Info
   //
   useEffect(() => {
-    if (
-      !poolData ||
-      !poolInfo?.totalPoolTokens ||
-      !poolInfoUser?.poolShares ||
-      !poolInfo?.totalLiquidityInOcean ||
-      !poolData?.baseTokenLiquidity ||
-      !asset?.chainId ||
-      !accountId ||
-      !poolInfoUser
-    )
-      return
-
-    // once we have poolData, we need to get user's pool shares (VL)
-    calculateSharesVL(
-      poolData.id,
-      poolData.baseToken.address,
-      poolInfoUser.poolShares,
-      asset.chainId
-    ).then((shares) => {
-      setUserPoolShares(shares)
-    })
-
-    // Pool share in %.
-    const poolShare = new Decimal(userPoolShares)
-      .dividedBy(new Decimal(poolInfo.totalLiquidityInOcean))
-      .mul(100)
-      .toFixed(2)
-
-    setUserHasAddedLiquidity(Number(poolShare) > 0)
-
-    const newPoolInfoUser = {
-      liquidity: new Decimal(userPoolShares), // liquidity in base token, values from from `calcSingleOutGivenPoolIn` method
-      poolShare
+    async function calculatePoolUserInfo() {
+      if (
+        !poolData ||
+        !poolInfo?.totalPoolTokens ||
+        !poolInfoUser?.poolShares ||
+        !poolData?.baseTokenLiquidity ||
+        !asset?.chainId
+      )
+        return
+
+      const userLiquidity = calcSingleOutGivenPoolIn(
+        poolData.baseTokenLiquidity,
+        poolData.totalShares,
+        poolInfoUser.poolShares
+      )
+
+      // Pool share in %.
+      const poolSharePercentage = new Decimal(poolInfoUser.poolShares)
+        .dividedBy(new Decimal(poolInfo.totalPoolTokens))
+        .mul(100)
+        .toFixed(2)
+
+      setUserHasAddedLiquidity(Number(poolSharePercentage) > 0)
+
+      const newPoolInfoUser = {
+        liquidity: userLiquidity,
+        poolshares: poolInfoUser.poolShares,
```

Comment:
it's `poolShares` in the typings, the new `const` should be assigned that typing above so this small typo should get visible in type checking then